paddle/fluid/memory/allocation/retry_allocator.cc

```diff
@@ -31,29 +53,29 @@ Allocation* RetryAllocator::AllocateImpl(size_t size) {
   // But it would add lock even when allocation success at the first time
   try {
     return alloc_func();
-  } catch (BadAlloc& bad_alloc) {
+  } catch (BadAlloc&) {
```

Comment:
Consider adding some logs when catching BadAlloc. Same as while loop beneath. It may be easier to debug in the future.

Reply:
Done.